pkg/dochandler/handler.go

```diff
@@ -139,7 +157,24 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
 		return nil, err
 	}
 
-	return pv.DocumentTransformer().TransformDocument(rm, getTransformationInfo(op.ID, false))
+	ti := make(protocol.TransformationInfo)
+	ti[document.PublishedProperty] = false
+
+	id := r.namespace + docutil.NamespaceDelimiter + op.UniqueSuffix
+
+	// if optional domain is specified we should set equivalent id with domain hint for interim/unpublished documents
+	if r.domain != "" {
+		id = r.namespace + docutil.NamespaceDelimiter + "interim" + docutil.NamespaceDelimiter + op.UniqueSuffix
+
+		equivalentID := r.namespace + docutil.NamespaceDelimiter +
+			"interim" + docutil.NamespaceDelimiter + r.domain + docutil.NamespaceDelimiter + op.UniqueSuffix
```

Comment:
I think the label portion (e.g., "interim:") is DID method specific.

Reply:
We should make this configurable/settable by the concrete implementation downstream project.

Reply:
It will only show up if domain is specified. Domain is not specified in other cases.

Reply:
I will make it configurable (same as domain).